react/features/app-settings/functions.native.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+
+import { Dimensions, Platform } from 'react-native';
+
+const IPHONE_OFFSET = 20;
+const IPHONEX_HEIGHT = 812;
+const IPHONEX_OFFSET = 44;
+const IPHONEX_WIDTH = 375;
+
+/**
+* FIXME: Settings screen uses a custom implementation of safe views
+* to avoid empty areas on the screen when there should be a header rendered.
+* In long term we need to use a generic solution to handle "safe view
+* areas" for various devices. It can be React 0.50's {@code SafeAreaView} or
+* a module like 'react-native-safe-area-view' or a custom implemented solution.
+*/
+
+/**
+* Determines if the device is an iPad or not.
+*
+* @returns {boolean}
+*/
+export function isIPad() {
```

Comment:
this looks like something which belongs in base/react/platform

Reply:
I think the comment applies this too: these pieces of logic (actually maybe the entire file) needs to be refactored when we have a generic SafeArea solution, and since no other feature uses this method yet, I'd leave it here now. Also, I'd still prefer a hardware-info based solution replacing these.

Reply:
There are other reasons why we might want to test for a specific device. This is related to the platform itself, so it certainly belongs there. If we end up removing it great, if not, it will already be in the right place. Since this code is being added now, let's add it in the right place.

Reply:
I would also vote for moving outside. Nobody will know that this thing exists here and will duplicate it eventually.